frontend/app/src/components/history/events/HistoryEventsView.vue

```diff
@@ -330,6 +331,16 @@ function onShowDialog(type: 'decode' | 'protocol-refresh'): void {
     set(protocolCacheStatusDialogOpen, true);
 }
 
+async function redecodePageTransactions(): Promise<void> {
+  const evmEvents = currentPageEvents.value.filter(isEvmEvent);
+  const { pullAndRedecodeTransaction, fetchUndecodedTransactionsStatus } = useHistoryTransactionDecoding();
+
+  for (const event of evmEvents) {
+    await pullAndRedecodeTransaction(toEvmChainAndTxHash(event));
+  }
```

Comment:
with this, the endpoint will be called many times.
Instead, make `pullAndRedecodeTransaction` accept array, (because the endpoint accept array too), so we don't need to call this many times